### PR TITLE
DeviceClient implemented.

### PR DESCRIPTION
**Pull request by SezginEge**

### CODE REVIEW

File: xomni-sdk-ts-preview/src/management/company/device-client.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            }, error);
+        }
+
+        private validateDevice(device: Models.Management.Company.Device) {
+            Xomni.Utils.Validator.isDefined("device", device);
+            Xomni.Utils.Validator.isDefined("description", device.Description);
+            Xomni.Utils.Validator.isGreaterThanOrEqual("relatedLicenceId", device.RelatedLicenceId, 1);
+        }
+
+        private convertToDate(deviceJson: any) {
```
Comment (ilkayilknur) [inline xomni-sdk-ts-preview/src/management/company/device-client.ts]: This method name doesn't reflect the implementation. convertToDevice would be better.

File: xomni-sdk-ts-preview/src/management/company/device-client.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                DeviceTypeDescription: deviceJson.DeviceTypeDescription,
+                DeviceTypeId: deviceJson.DeviceTypeId,
+                ExpirationDate: deviceJson.ExpirationDate ? new Models.UTCDate(deviceJson.ExpirationDate): null,
+                RelatedLicenceId: deviceJson.RelatedLicenceId,
+                RelatedLicenceName: deviceJson.RelatedLicenceName
+            };
+            return device;
+        }
+
+        private convertListToDate(list: any) {
```
Comment (ilkayilknur) [inline xomni-sdk-ts-preview/src/management/company/device-client.ts]: Same here.

Comment (SezginEge) [inline xomni-sdk-ts-preview/src/management/company/device-client.ts]: @ilkayilknur Is convertListToDeviceList appropriate for you?

Comment (ilkayilknur) [inline xomni-sdk-ts-preview/src/management/company/device-client.ts]: convertToDeviceList is better.

**Pull request merged**